### ISSUE TITLE
Application Insights Instace is not getting provisioned

### ISSUE THREAD

**Issue by indcoder** [Azure/static-web-apps#673]
**Describe the bug**

Have a managed function called GetRoles and I'm trying to provision an Application Insight instance to monitor it. And SWA hosting plan is on Standard Plan.

**To Reproduce**

Followed step by step the below resource,

https://docs.microsoft.com/en-us/azure/static-web-apps/monitor

**Expected behavior**
Application Insight should be provisioned for the Azure SWA. 

**Screenshots**

Provisioning at start

<img width="590" alt="image" src="https://user-images.githubusercontent.com/2099300/147580464-7a8b13cf-6d95-4ddb-a6a6-30906475d5bb.png">

No AI resources is created 
<img width="446" alt="image" src="https://user-images.githubusercontent.com/2099300/147581002-fdf88a43-695a-4a52-b0cc-b56c46124721.png">

Also going back to Application Insight in SWA blade _suggests_ that AI instance was not created because the radio button still defaults to no for Enable Application Insights

<img width="436" alt="image" src="https://user-images.githubusercontent.com/2099300/147581396-a1358692-0c20-4588-bccf-483012263c08.png">

And going back to Application Insight 

**Device info (if applicable):**

**Additional context**

**Comment by staplJason** [comment]
I had the same issue.  I used the info in this link to get my demo working: 
https://www.aaron-powell.com/posts/2020-08-12-getting-logs-from-static-web-apps-apis/

**Comment by anthonychu** [comment]
@shimedh @simonaco Can you please investigate?

**Comment by simonaco** [comment]
@indcoder @staplJason thanks for reporting this issue. I wasn't able to reproduce it with an older app nor with a new app. did you create the app insights instance separately or directly from the static web apps UI? can you share your hostname or github repo so we can investigate?

**Comment by indcoder** [comment]
Thanks @simonaco for looking into this.  The screenshot from my original post shows the screenshot that I used to do this from the Azure portal [Static Web Apps UI]. 

Host name : https://ashy-cliff-098028800.azurestaticapps.net
GitHub repo: https://github.com/indcoder/Visage

**Comment by staplJason** [comment]
This was on a brand new account and creating a static web app was the first thing I tried.
I just created another static app and everything worked as expected.

A few things I remember as being different: 
* My first static app never had any stats on the overview screen until I got the Application Insights working using the info in the link provided earlier.
* Before the first static app I didn't  have a "DefaultWorkspace/Log Analytics workspace" Not sure when it was created (Maybe while following the examples in the link attached earlier in this issue)
![image](https://user-images.githubusercontent.com/38960789/148157704-6c44e365-2baf-449a-9292-7098f828dd42.png)

* For the 2nd static app --By the time I added an api and committed the overview screen already had request stats *Before* enabling application insights.

**Comment by thehitchiker** [comment]
I have this exact issue on a new subscription with the swa free tier. The option is enabled so I'm assuming AI is available on the free tier. I haven't read otherwise yet. No errors messages but nothing is created. I've upgraded the subscription in case it was that but still no luck. 

Perhaps I have to create an AI account manually first? 

I just did exactly that and all works as expected.

**Comment by indcoder** [comment]
@thehitchiker Correct, I was in contact with the AI team to conclusively prove this, and they have already got the patch in.
